src/main/groovy/de/tracetronic/jenkins/plugins/ecutestexecution/clients/RestApiClient.groovy

```diff
@@ -29,22 +33,28 @@ interface RestApiClient {
 
     /**
      * This method performs the package check for the given test package or project via REST api.
+     * The method will abort upon a thread interruption signal used by the TimeoutControllerToAgentCallable to handle the
+     * timeout from outside this class.
+     * {@see de.tracetronic.jenkins.plugins.ecutestexecution.security.TimeoutControllerToAgentCallable}
      * @param testPkgPath the path to the package or project to be checked
-     * @param timeout Time in seconds until the check package execution will be aborted
      * @return CheckPackageResult with the result of the check
-     * @throws ApiException on error status codes
-     * @throws TimeoutException on timeout exceeded
+     * @throws ApiException on error status codes (except 409 (busy) where it will wait until success or timeout)
+     * @throws TimeoutException during api calls if executionTimedOut is set to true by
```

Comment:
~executionTimedOut~ -> execution exceeds timeout
Are you finished the sentence?